### PR TITLE
Village Variants

### PR DESCRIPTION
**Pull request by WenXin20**
- Adds new savanna, snowy, & taiga village variants.
- Plains & desert villages are now more diverse.
- Adds new PokeCenter and Gym variants for each biome type.
- Cats & Iron Golems now spawn in these villages.
- Ocean village has more roads, but houses are not yet spawning correctly.
- Fix stonehenge ruins floating on water.

### CODE REVIEW

**Comment by Thutmose** [comment]
you might want to check the commit, it changes some of the ant-ai related classes, seems like it reverts some of the recent changing in them

File: src/main/java/pokecube/core/ai/brain/RootTask.java [hunk 1]
```diff
@@ -106,12 +106,6 @@ protected void setWalkTo(final Entity mobIn, final double speed, final int dist)
 
     protected void setWalkTo(final WalkTarget target)
     {
-        if (!(target.getTarget() instanceof EntityPosWrapper) && target != null)
-        {
-            final boolean inRange = target.getTarget().getPos().isWithinDistanceOf(this.entity.getPositionVec(), target
-                    .getDistance());
-            if (inRange) return;
-        }
         this.entity.getBrain().setMemory(MemoryModules.WALK_TARGET, target);
```
Comment (Thutmose) [inline src/main/java/pokecube/core/ai/brain/RootTask.java]: this probably shouldn't be reverted

Comment (WenXin20) [reply src/main/java/pokecube/core/ai/brain/RootTask.java]: See if my "Revert the revert of the 1.16 branch" fixed it"?

Comment (Thutmose) [reply src/main/java/pokecube/core/ai/brain/RootTask.java]: No, now it has reverted even more changes from the main branch

**Comment by WenXin20** [comment]
Closed in favor of a fresh pull request

**Pull request closed without merging**